### PR TITLE
Add customer examples in C++

### PR DESCRIPTION
**Pull request by jtibshirani**
Adding some recent examples inspired by a customer issue we recently debugged
around qualified C++ names like `tensorflow::OpRegistry`. This will prevent
regressions and let us do more error analysis around C++ and python.

### CODE REVIEW

**Comment by jtibshirani** [comment]
Notes:
* I tried to rerun the embeddings eval, but I couldn't embed the whole tensorflow repo before getting rate limited 🤔 
* There's a new column `Source` which tracks whether an example came from a customer, dogfooding, or synthetic (meaning I just made it up because I thought it was a nice example!)
* I'll also update `sourcegraph/dev-private` to pull in `pinned-tensorflow` repo by default

File: golden_queries/queries.csv [hunk 1]
```diff
@@ -1,4 +1,4 @@
-Question,Class,Repo,File type,Best files,Other files,Provenance
```
Comment (janhartman) [inline golden_queries/queries.csv:1]: I admit I had to refresh my knowledge on what provenance means 🙈 good change!

**Review by janhartman — APPROVED** [review]
Nice! Not surprised that you weren't able to embed the full TF repo, it's absolutely massive...

**Review by stefanhengl — APPROVED** [review]

File: golden_queries/queries.csv [hunk 2]
```diff
@@ -8,6 +8,8 @@ Examples of useFeatureFlag used in sourcegraph,Find symbol,sourcegraph,Typescrip
...
+where is tensorflow::OpRegistry defined?,Find symbol,tensorflow,C++,tensorflow/core/ir/tf_op_registry.h,,Customer
+find examples ReadBinaryProto on graph def,Find symbol,tensorflow,C++,tensorflow/tools/proto_splitter/cc/graph_def_splitter_test.cc,,Synthetic
```
Comment (jtibshirani) [inline golden_queries/queries.csv:12]: @stefanhengl here's another example where not implementing BM25 hurts us. The 'examples' term seems to dominate the results.

**Pull request merged**